### ISSUE TITLE
Error spawning PHP: spawn EACCES

### ISSUE THREAD

**Issue by tjdraper** [felixfbecker/vscode-php-intellisense#25]
I’m running into this error and I’m not quite sure what to do about it. A search through the issues reveals nothing so could be unique to my setup?

![screen shot 2016-11-02 at 10 27 44 am](https://cloud.githubusercontent.com/assets/3803475/19935170/19f8580c-a0e7-11e6-8fa6-f80da3da58b1.png)

**Comment by felixfbecker** [comment]
Check the permissions on the PHP executable

**Comment by tjdraper** [comment]
0755

**Comment by Kejjden** [comment]
I got this problem when I had the wrong php.executablePath setting.
I was using /path/to/php/bin/ and changed it to /path/to/php/bin/php.

**Comment by felixfbecker** [comment]
@tjdraper Are you using this setting? if yes, what is it set to? This is very certainly a problem with your setup, not a bug in the extension

**Comment by felixfbecker** [comment]
Closing for lack of response

**Comment by hsnyc** [comment]
Hi Felix, I have encounter this issue using a MAC. I have set php.executablePath to "php.executablePath": "/Applications/MAMP/conf/php7.2.8/" and I get this error. In this dir there is only two files: php.ini and pear.conf. Any suggestions are appreciated. Thank you.

**Comment by felixfbecker** [comment]
`php.executablePath` needs to point to a binary, not a directory.

**Comment by hoangnamitc** [comment]
![image](https://user-images.githubusercontent.com/7042462/57573985-2f8b1200-745b-11e9-928b-af327f265521.png)

hi guy, my dir php only image , How to set path for php.executablePath

tks

**Comment by sunverwerth** [comment]
Try typing `which php` in the terminal,

**Comment by Blair2004** [comment]
@sunverwerth this helped me to make this Extension work on Remote Server (Remote Development with VSCode). Thanks

**Comment by OmarMakled** [comment]
Try this

VS: Version 1.32.1 (1.32.1)
OS: macOS 10.14.5 
```
  "php.executablePath": "/usr/local/opt/php@7.2/bin/php",
  "php.validate.executablePath": "/usr/local/opt/php@7.2/bin/php",
```

**Comment by wltrallen2** [comment]
@sunverwerth Thank you! I'd been searching for the solution for quite a while now and this was super helpful. Thank you! (Don't know why I didn't think of it. LOL. It was kind of a face palm moment.)

**Comment by CarlosArceo10** [comment]
@sunverwerth Thanks, your advice helped me

**Comment by devgac** [comment]
I had a similar issue on Mac with php running in a container.
I created an executable /usr/local/bin/php which ran a docker exec command basically. Initially the rights were 755 and owner/group was "root / admin"
 
The error code was : **"Error spawning PHP: spawn -8"**

I managed to fix this by changing ownership on the executable (assigned to my user). This is strange because executable permission was enabled for "others"
